### PR TITLE
Max limit - Backend

### PR DESCRIPTION
**Pull request by SaladRaider**
### What does this PR do?
Sets a max limit on the number of project applications and accepted volunteers.

### Status
READY

### Todos
- Implement frontend

### Screenshots
Screen Shot 2019-03-15 at 12 20 19 PM

CC: @steveshi7

### CODE REVIEW

File: app/controllers/api/applications_controller.rb [hunk 1]
```diff
@@ -52,11 +57,12 @@ def update
 
   def decide
     decision = params[:decision]
-    puts("DECISION")
-    puts(decision)
 
     begin
       application = Application.find(params[:id])
+      if decision == 2 and application.project.has_max_users
```
Comment (steveshi7) [inline app/controllers/api/applications_controller.rb]: don't refer to models by their ids correctly, these are not static

File: db/schema.rb [hunk 2]
```diff
@@ -104,6 +104,8 @@
     t.text "question2"
     t.text "question3"
     t.bigint "project_type_id"
+    t.integer "max_applications", default: 2
```
Comment (steveshi7) [inline db/schema.rb]: rename this as application_limit, user_limit, also in your project/rb

File: app/controllers/api/applications_controller.rb (hunk 1 above)
Comment (SaladRaider) [reply app/controllers/api/applications_controller.rb]: do you want me to refer to it by their string name?

Comment (SaladRaider) [reply app/controllers/api/applications_controller.rb]: PR #59 changes this to be an enum, so i'll change this on rebase once that PR gets merged in.

File: app/controllers/api/applications_controller.rb [hunk 3]
```diff
@@ -52,11 +57,12 @@ def update
 
   def decide
     decision = params[:decision]
-    puts("DECISION")
-    puts(decision)
 
     begin
       application = Application.find(params[:id])
+      if decision == 2 and application.project.reached_user_limit
```
Comment (steveshi7) [inline app/controllers/api/applications_controller.rb]: still not fixed

**Pull request merged**